File: src/1385-Find-the-Distance-Value-Between-Two-Arrays/1385.js
var findTheDistanceValue = function(arr1, arr2, d) {
    arr2.sort((a, b) => a - b);
    
    let check = function(x) {
        let l = 0, r = arr2.length - 1;
        while (l < r) {
            let mid = (l + r) >> 1;
            if (Math.abs(arr2[mid] - x) <= d) return false;
            if (arr2[mid] > x) r = mid;
            else l = mid + 1;
        }
        return Math.abs(arr2[l] - x) > d;
    }
    
    let res = 0;
    for (let x of arr1) {
        if (check(x)) res++;
    }
    return res;
};

File: src/1385-Find-the-Distance-Value-Between-Two-Arrays/1385.java
class Solution {
    public int findTheDistanceValue(int[] arr1, int[] arr2, int d) {
        Arrays.sort(arr2);
        
        int res = 0;
        this.d = d;
        for (int x : arr1) {
            if (check(arr2, x)) res++;
        }
        return res;
    }
    
    private int d;
    
    private boolean check(int[] arr2, int x) {
        int l = 0;
        int r = arr2.length - 1;
        while (l < r) {
            int mid = (l + r) >> 1;
            if (Math.abs(arr2[mid] - x) <= d) return false;
            if (arr2[mid] > x) r = mid;
            else l = mid + 1;
        }
        return Math.abs(arr2[l] - x) > d;
    }
}